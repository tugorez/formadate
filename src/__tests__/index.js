import mockdate from 'mockdate';
import formadate from '../index';

mockdate.set(new Date('2017-03-13T22:12:39.583Z'));

describe('formadate', () => {
  it('should render the time in am/pm if the date is today', () => {
    const date = new Date('2017-03-13T21:04:36.000Z');
    const formated = formadate(date);
    expect(formated).toMatchSnapshot();
  });

  it('should render the day if the date is in this week', () => {
    const date = new Date('2017-03-07T05:00:00.000Z');
    const formated = formadate(date);
    expect(formated).toMatchSnapshot();
  });

  it('should render the month and day if date is in this year', () => {
    const date = new Date('2017-01-01T05:00:00.000Z');
    const formated = formadate(date);
    expect(formated).toMatchSnapshot();
  });

  it('should render the month day and year otherwise', () => {
    const date = new Date('2016-12-31T05:00:00.000Z');
    const formated = formadate(date);
    expect(formated).toMatchSnapshot();
  });
});
