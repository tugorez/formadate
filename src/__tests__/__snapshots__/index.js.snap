// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formadate should render the day if the date is in this week 1`] = `"martes a las 12:00 am"`;

exports[`formadate should render the month and day if date is in this year 1`] = `"ene. 1 a las 12:00 am"`;

exports[`formadate should render the month day and year otherwise 1`] = `"dic. 31 2016 a las 12:00 am"`;

exports[`formadate should render the time as calendar 1`] = `"Today at 4:04 PM"`;

exports[`formadate should render the time from now 1`] = `"an hour ago"`;

exports[`formadate should render the time in am/pm if the date is today 1`] = `"hoy a las 4:04 pm"`;
